### ISSUE TITLE
[Rule request] Internal as private

### ISSUE THREAD

**Issue by kimdv** [realm/SwiftLint#2503]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:
When a method is internal but not used outside of class the rule should warn to make it private. 

_Provide several examples of what _would_ and _wouldn't_ trigger violations._

**Nontrigger**
```swift
 """
class Foo {
    func bar() { }
}

let foo = Foo()
foo.bar()
"""
```

**Trigger**
```swift
 """
class Foo {
    func bar() { }
    private func baz() {
        bar()
    }
}

"""
```

_Should the rule be opt-in or enabled by default? Why?_
Because it is analyzer rule it should be opt-in.

**Comment by kimdv** [comment]
I have tried to implement this with inspiration from `UnusedPrivateDeclarationRule`

But I'm stuck on how to solve remove all the usages from the class `Foo` from `referencedUSRs`

**Comment by jpsim** [comment]
I think this will require multifile support.

**Comment by kimdv** [comment]
Is that something i can help with or is that not in the pipeline yet?

**Comment by kimdv** [comment]
I see that you started discussing in #2431

**Comment by georgealegre** [comment]
I would love to have this.
